### PR TITLE
Avoid throwing error when controller kind is unknown

### LINKED ISSUE #3251: Dashboard errors if pod's owner reference is not supported
##### Steps to reproduce

The pod is created as:
`podObject.OwnerReferences = []meta.OwnerReference{*meta.NewControllerRef(configMapObject, ConfigMapGroupVersionKind)}`

##### Environment
```
Installation method: 
Kubernetes version:
Dashboard version: v1.8.1
Commit: 7b3553aecf28224ef7275e157b5bba7dd23f436a
```

##### Observed result
Dashboard reported Internal Server Error (500):
```
Unknown reference kind ConfigMap

```

##### Comments

By setting the owner is also a controller, the Pod functions normally, except for its detail cannot be viewed from kube dashboard, and the dashboard shows error:
"Unknown reference kind ConfigMap".

Seems this is related to https://github.com/kubernetes/dashboard/issues/3158

Please help :)

### PR DESCRIPTION
**Pull request by maciaszczykm**
Fixes #3251.

### CODE REVIEW

File: src/app/backend/resource/pod/detail.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ type PodDetail struct {
 	NodeName                  string                                          `json:"nodeName"`
 	RestartCount              int32                                           `json:"restartCount"`
 	QOSClass                  string                                          `json:"qosClass"`
-	Controller                controller.ResourceOwner                        `json:"controller"`
+	Controller                *controller.ResourceOwner                       `json:"controller,omitempty"`
```
Comment (jeefy) [inline src/app/backend/resource/pod/detail.go]: I think this is breaking the tests in detail_test.go

See https://travis-ci.org/kubernetes/dashboard/jobs/496515775#L364

Comment (maciaszczykm) [reply src/app/backend/resource/pod/detail.go]: Sure, didn't check it before. Thanks :)

**Comment by maciaszczykm** [comment]
@jeefy PTAL

**Pull request merged**